src/graph/executor/StorageAccessExecutor.cpp

```diff
@@ -148,5 +149,85 @@ StatusOr<std::vector<Value>> StorageAccessExecutor::buildRequestListByVidType(It
   return internal::buildRequestList<std::string>(space, exprCtx, iter, expr, dedup, isCypher);
 }
 
+bool StorageAccessExecutor::hasSameEdge(const std::vector<Value> &edgeList, const Edge &edge) {
+  for (auto &leftEdge : edgeList) {
+    if (!leftEdge.isEdge()) {
+      continue;
+    }
+    if (edge.keyEqual(leftEdge.getEdge())) {
+      return true;
+    }
+  }
+  return false;
+}
+
+folly::Future<std::vector<Value>> StorageAccessExecutor::getProps(
+    const std::vector<Value> &vids, const std::vector<VertexProp> *vertexPropPtr) {
+  nebula::DataSet vertices({kVid});
+  vertices.rows.reserve(vids.size());
+  for (auto &vid : vids) {
+    vertices.emplace_back(Row({vid}));
+  }
+  StorageClient *storageClient = qctx_->getStorageClient();
+  StorageClient::CommonRequestParam param(qctx_->rctx()->session()->space().id,
+                                          qctx_->rctx()->session()->id(),
+                                          qctx_->plan()->id(),
+                                          qctx_->plan()->isProfileEnabled());
+  return DCHECK_NOTNULL(storageClient)
+      ->getProps(
+          param, std::move(vertices), vertexPropPtr, nullptr, nullptr, false, {}, -1, nullptr)
+      .via(runner())
+      .thenValue([this](PropRpcResponse &&resp) {
+        addStats(resp);
```

Comment:
add `memory::MemoryCheckGuard guard;`
reference to TraverseExecutor.cpp for memory check usage

Reply:
👌